code/game/objects/structures/platforms.dm

```diff
@@ -79,6 +92,49 @@
 			SPAN_WARNING("You finish dragging \the [user.pulling] onto \the [src]."))
 			user.pulling.forceMove(move_to_turf)
 
+/obj/structure/platform/attack_alien(mob/living/carbon/xenomorph/user)
+	if(stat & BROKEN)
+		to_chat(user, SPAN_WARNING("Its already destroyed!"))
+		return XENO_NO_DELAY_ACTION
+
+	if(stat & indestructible)
+		to_chat(user, SPAN_WARNING("Its too strong for us!"))
+		return XENO_NO_DELAY_ACTION
+
+	if(user.action_busy)
+		return XENO_NO_DELAY_ACTION
+	if(user.a_intent != INTENT_DISARM)
+		user.set_interaction(src)
+		tgui_interact(user)
+		return XENO_ATTACK_ACTION
+	user.visible_message(SPAN_WARNING("[user] begins to lean against [src]."), \
+	SPAN_WARNING("You start to stomp and pressure [src]."), null, 5, CHAT_TYPE_XENO_COMBAT)
+	playsound(loc, creaking_sound, 30, 1)
+
+	var/shove_time = 90
+	if(istype(user,/mob/living/carbon/xenomorph/lesser_drone))
+		shove_time = 300 //They are very light and weak, thats why they will take 30s to break one edge.
+	if(istype(user,/mob/living/carbon/xenomorph/defender))
+		shove_time = 70
+	if(istype(user,/mob/living/carbon/xenomorph/burrower))
+		shove_time = 50
+	if(istype(user,/mob/living/carbon/xenomorph/boiler))
+		shove_time = 40
+	if(user.mob_size >= MOB_SIZE_BIG)
+		shove_time = 30
+	if(istype(user,/mob/living/carbon/xenomorph/queen))
+		shove_time = 15 //Queen is just too fat for it to keep her weight.
```

Comment:
if you want for all xeno to have diffrent time to break the ledge, add a var on xeno type, an easier approach would be to look at the mob size because you dont need to add var at xeno type. But this is disgusting.